sdk-wrapper/src/main/java/com/snowflake/ingest/sdk/server/StreamingIngestJavaService.java

```diff
@@ -232,6 +232,32 @@ public HttpResponse insertRow(
             "offsetToken", offsetToken));
   }
 
+  @Post("/clients/{clientId}/channels/{channelName}/insertRows")
+  public HttpResponse insertRows(
+      @Param("clientId") String clientId,
+      @Param("channelName") String channelName,
+      @RequestObject java.util.List<Map<String, Object>> rows,
+      @Param("offsetToken") String offsetToken)
+      throws Exception {
+    SnowflakeStreamingIngestChannel channel = getChannelOrThrow(clientId, channelName);
+
+    // Insert all rows with the same offset token
+    for (Map<String, Object> row : rows) {
+      InsertValidationResponse response = channel.insertRow(row, offsetToken);
+      if (response.hasErrors()) {
+        throw response.getInsertErrors().get(0).getException();
```

Comment:
When you create the channel, you can specify the ON_ERROR option and if we do ABORT, it will throw the exception for you.